Assignment10.txt

```diff
@@ -0,0 +1,287 @@
+		// TODO Auto-generated method stub
+		List<Function> functions=new ArrayList<Function>(role.getFunctions());
+		
+		List<Map<String, Object>> batchValues = new ArrayList<>(functions.size());
+		for (Function f : functions) {
+		    batchValues.add(
+		            new MapSqlParameterSource("RI", role.getRoleId())
+		                    .addValue("FI", f.getFunctionId())
+		                    .addValue("STATUS", f.getStatus())
+		                    .getValues());
+		}
+
+		template.batchUpdate(QueryRole.INSERT_ROLE_FUNCTION,
+		                           batchValues.toArray(new Map[functions.size()]));
+		return findById(role);
+	}
+*Explanation:
+Input:Role object as method parameter.
+mapParams:RI:gets Role Id from role Object
+		 :FI:gets Function id from Function Object.
+		 :STATUS:gets status from Role and Function Object.
+query:QueryRole.INSERT_ROLE_FUNCTION
+queryAction:Insert status into  TBL_ROLE_FUNCTION table with Function id and Role Id.
+return value:Role
+*/
+6.public Role updateAssignedFunctionsStatusById(Role role) :
+{
+		// TODO Auto-generated method stub
+		Map<String, Integer>[] maps = new HashMap[role.getFunctions().size()];
+		int index = 0;
+		for (Function f : role.getFunctions()) {
+			Map<String, Integer> dataMap = new HashMap<>();
+			dataMap.put("RI", role.getRoleId());
+			dataMap.put("FI", f.getFunctionId());
+			dataMap.put("STATUS", f.getStatus());
+			maps[index] = dataMap;
+			index++;
+		}
+		if (maps.length > 0) {
+			int[] batchSize = template.batchUpdate(QueryRole.UPDATE_STATUS_BY_ROLE_ID_FUNCTION_ID, maps);
+		}
+		return findById(role);
+	}
+/*Explanation:
+Input:Role object as method parameter.
+mapParams:RI:gets RoleId from role Object
+		 :FI:gets FunctionId from Function Object
+		 :STATUS:gets Status from Role and Function Object
+query:QueryRole.UPDATE_STATUS_BY_ROLE_ID_FUNCTION_ID
+queryAction:It will update the status in the TBL_ROLE_FUNCTION table with the RoleId and FunctionId.
+return value:Role
+*/
+7.public Role findById(Role role):
+{
+		// TODO Auto-generated method stub
+		MapSqlParameterSource source=new MapSqlParameterSource();
+		source.addValue("RI", role.getRoleId());
+		role=template.queryForObject(QueryRole.GET_ROLE_BY_ID, source, new RoleRowMapper());
+		role.setFunctions(new HashSet<>(getAssignedFunctionsById(role)));
+		return role;
+	}
+*Explanation:
+Input:Role object as method parameter.
+mapParams:RI:gets RoleId from role Object.
+query:QueryRole.GET_ROLE_BY_ID
+queryAction:It will retrieve all the dats fromm the TAB_ROLE table by the RoleId.
+RowMapper:it uses RoleRowMapper which contains logic to retrieve Role with resuletset.
+return value:Role
+*/
+8.public List<Role> findAll()
+{
+		// TODO Auto-generated method stub
+		return template.query(QueryFunction.SELECT_ALL, new RoleRowMapper());
+	}
+*Explanation:
+Input:Role object as method parameter.
+query:QueryFunction.SELECT_ALL
+queryAction:It will retrieve all the datas From TBL_FUNCTION table
+return value:List
+*/
```

Comment:
@PallaviMonamitech  with this we finished the dao layer. What about serviceLayer and controllerLayer.
Can you please add serviceLayer in another file into this repository